### PR TITLE
TEP-0110: Decouple Catalog Organization and Resource Reference

### PR DESCRIPTION
**Pull request by jerop**
As described in TEP-0003: Tekton Catalog Organization,
the Tekton Catalog is a collection of blessed Tekton resources
(`Tasks` and `Pipelines`) that can be used with any system that
supports the Tekton API. On the hand, the Tekton Hub to be
able to search for resources.

Today, versioning of resources in the Catalog is directory-based
and there is tight coupling between the organization and how users
fetch resources. Therefore, it is challenging to change the
organization of the Catalog without breaking users.

This TEP proposes to make the *Hub* and any other **official**
mean of consuming Tekton resources from the community
decoupled from where we author them. In a gist, this aims to make
the repository `tektoncd/catalog` an implementation detail.
One approach is to support long-term and non-changing URI to refer
to resources via `https://` endpoints, such as the Hub or OCI images.
Another approach is to support remote resolution, through Hub Resolver
and Git Resolver, that does not depend on the organization of the
Catalogs.

### CODE REVIEW

**Comment by jerop** [comment]
@vdemeester please take a look 🙏🏾

File: teps/0110-decouple-catalog-organization-and-reference.md [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+resources (`Tasks` and `Pipelines`) that can be used with any system that supports the Tekton API. On the hand, the
+[Tekton Hub][hub] to be able to search for resources.
+
+Today, versioning of resources in the Catalog is directory-based and there is tight coupling between the organization
+and how users fetch resources. Therefore, it is challenging to change the organization of the Catalog without breaking
+users.
+
+This TEP proposes to make the *Hub* and any other **official** mean of consuming Tekton resources from the community
+decoupled from where we author them. In a gist, this aims to make the repository `tektoncd/catalog` an implementation
+detail. One approach is to support long-term and non-changing URI to refer to resources via `https://` endpoints, such
```
Comment (vdemeester) [inline teps/0110-decouple-catalog-organization-and-reference.md:45]: nit: `https://` endpoint only applies to part of the URI (OCI images do not really have that).

File: teps/0110-decouple-catalog-organization-and-reference.md [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+structure of the data there requires users to process the data first before applying the resources to their clusters.
+The new HTTP endpoint mentioned above should serve the *yaml* files that users can easily install resources to their
+clusters, similarly the current experience using GitHub HTTP endpoint:
+
+```shell
+# Current: using GitHub path in existing Tekton Catalog organization
+kubectl apply -f https://raw.githubusercontent.com/tektoncd/catalog/main/task/buildpacks/0.4/buildpacks.yaml
+
+# Proposal: using the HTTP Endpoint from Tekton Catalog - should serve same yaml as the one above
+kubectl apply -f https://hub.tekton.dev/tekton//task/buildpacks/0.4
```
Comment (vdemeester) [inline teps/0110-decouple-catalog-organization-and-reference.md:112]: nit: extra `/`

**Review by vdemeester — APPROVED** [review]

**Comment by jerop** [comment]
Related PR: https://github.com/tektoncd/community/pull/663

**Pull request merged**